### PR TITLE
attempt to pipe all site provisioner output to the log files

### PR DESCRIPTION
**Pull request by tomjn**
## Summary:

This attempts to pipe all site template provisioning to the log file, bypassing the terminal, needs testing and polish

## Checks

 - [ ] I've tested this PR with Vagrant **vXX** and VirtualBox **vXX** on **Operating System**
 - [x] This PR is for the `develop` branch not the `master` branch
 - [x] I've updated the changelog
 - [ ] This PR is complete and ready for review

### CODE REVIEW

**Comment by tomjn** [comment]
This is functional, the final part is that it needs to output in red if the provisioner script failed. The return code doesn't seem to work for that, so further investigation is needed into what to do next

File: provision/provision-site.sh [hunk 1]
```diff
@@ -111,7 +111,7 @@ if [[ true == $SKIP_PROVISIONING ]]; then
   return
 fi
 
-echo -e "${BLUE}Running provisioner for site ${SITE}${CRESET}"
+echo -e "${GREEN}Running provisioner for site ${SITE}${CRESET}"
```
Comment (msaggiorato) [inline provision/provision-site.sh]: Shouldn't we standarize colors like this?

Green: Success
Red: Error
Yellow: Warning
Blue: Info

That's the reason I used blue.

**Comment by Mte90** [comment]
Looking at https://medium.com/@dirk.avery/the-bash-trap-trap-ce6083f36700 we can add at the top on this provisioner:

```
set -e
trap 'catch $? $LINENO' EXIT

catch() {
  echo "catching!"
  if [ "$1" != "0" ]; then
    # error handling goes here
    echo "Error $1 occurred on $2"
  fi
}
```
We can evaluate `set -ex` in case.

**Comment by tomjn** [comment]
I changed the blue to green because it was almost impossible to read dark
blue on black for me

**Comment by Mte90** [comment]
Right now to understand the terminal background color there isn't a standard way but there are workaround (https://github.com/rocky/bash-term-background/blob/master/term-background.sh).
I think that for the shell color output we can do another pr to address better those values.

**Comment by msaggiorato** [comment]
Maybe we can change to cyan?

**Comment by tomjn** [comment]
I'm not sure we can rely on that, lets just use green red and yellow for now

**Comment by Mte90** [comment]
About the strict execution there is a new pr so better focus on the pipe of the log

**Comment by Mte90** [comment]
I tested right now and there are no issues:
```
==> default: Running provisioner: site-woocommerce (shell)...
    default: Running: /tmp/vagrant-shell20191023-22496-1u3tb3g.sh
    default: Running provisioner for site woocommerce
    default: 
    default: Updating woocommerce in /srv/www/woocommerce...
    default: Searching for site template provisioner, vvv-init.sh
    default: Found vvv-init.sh at /srv/www/woocommerce/provision/vvv-init.sh
    default: Script output will be logged to: /var/log/provisioners/20191023-123027/provisioner-site-woocommerce.log
```

**Comment by Mte90** [comment]
I am getting the error printed like: `default: Deprecation warning: require-dev.mikey179/vfsStream is invalid, it should not contain uppercase characters. Please use mikey179/vfsstream instead. Make sure you fix this as Composer 2.0 will error.` but the majority of stuff is in the log itself.

**Comment by tomjn** [comment]
I’d like to know what the cause of that is before merging. It might be
nothing but better safe than sorry, sure provisioning is at the core of
things so I don’t want to be taking chances

**Comment by Mte90** [comment]
That error is something based on the develop provisioner but it's there since months because of a dependence of phpunit that is abandoned.
I written here as example that is tracking the errors also the "fake positive" and this is good.

**Comment by tomjn** [comment]
Is there a ticket or issue for PHPUnit?

**Comment by Mte90** [comment]
There is of this library https://github.com/bovigo/vfsStream/issues/173 seems that one of the dependence of composer in develop has this but is a non-issue.
Doesn't involve this pr anyway.

**Pull request merged**